pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/JUnitSpellingRule.java

```diff
@@ -5,13 +5,19 @@
 
 package net.sourceforge.pmd.lang.java.rule.errorprone;
 
+import net.sourceforge.pmd.lang.java.ast.ASTCompilationUnit;
 import net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration;
 import net.sourceforge.pmd.lang.java.rule.AbstractJUnitRule;
 
 public class JUnitSpellingRule extends AbstractJUnitRule {
 
     @Override
     public Object visit(ASTMethodDeclaration node, Object data) {
+        ASTCompilationUnit acu = node.getFirstParentOfType(ASTCompilationUnit.class);
+        if (isJUnit5Class(acu) || isJUnit4Class(acu)) {
```

Comment:
If using the protected fields from the base class, you don't need to search here again for the compilation unit and call again the methods - that was already done....

Reply:
You're right